dissect/target/plugins/apps/webserver/apache.py

```diff
@@ -187,66 +216,153 @@ class ApachePlugin(WebserverPlugin):
 
     def check_compatible(self) -> None:
-        if not len(self.access_log_paths) and not len(self.error_log_paths):
+        if not len(self.access_paths) and not len(self.error_paths):
             raise UnsupportedPluginError("No Apache directories found")
 
-    @plugin.internal
-    def get_log_paths(self) -> tuple[list[Path], list[Path]]:
-        """
-        Discover any present Apache log paths on the target system.
+        if self.target.os in [OperatingSystem.CITRIX, OperatingSystem.FORTIOS]:
+            raise UnsupportedPluginError("Use other Apache plugin instead")
+
+    def find_logs(self) -> tuple[list[Path], list[Path]]:
+        """Discover any present Apache log paths on the target system.
 
         References:
+            - https://httpd.apache.org/docs/2.4/logs.html
+            - https://httpd.apache.org/docs/2.4/mod/mod_log_config.html
             - https://www.cyberciti.biz/faq/apache-logs/
             - https://unix.stackexchange.com/a/269090
         """
 
-        access_log_paths = set()
-        error_log_paths = set()
-
         # Check if any well known default Apache log locations exist
         for log_dir, log_name in itertools.product(self.DEFAULT_LOG_DIRS, self.ACCESS_LOG_NAMES):
-            access_log_paths.update(self.target.fs.path(log_dir).glob(f"{log_name}*"))
+            self.access_paths.update(self.target.fs.path(log_dir).glob(f"*{log_name}*"))
 
         for log_dir, log_name in itertools.product(self.DEFAULT_LOG_DIRS, self.ERROR_LOG_NAMES):
-            error_log_paths.update(self.target.fs.path(log_dir).glob(f"{log_name}*"))
+            self.error_paths.update(self.target.fs.path(log_dir).glob(f"*{log_name}*"))
 
         # Check default Apache configs for CustomLog or ErrorLog directives
         for config in self.DEFAULT_CONFIG_PATHS:
             if (path := self.target.fs.path(config)).exists():
-                for line in path.open("rt"):
-                    line = line.strip()
+                self._process_conf_file(path)
 
-                    if not line or ("CustomLog" not in line and "ErrorLog" not in line):
-                        continue
+        # Check all .conf files inside the server root
+        if self.server_root:
+            for path in self.server_root.rglob("*.conf"):
+                self._process_conf_file(path)
 
-                    if "ErrorLog" in line:
-                        set_to_update = error_log_paths
-                        pattern_to_use = RE_CONFIG_ERRORLOG_DIRECTIVE
-                    else:
-                        set_to_update = access_log_paths
-                        pattern_to_use = RE_CONFIG_CUSTOM_LOG_DIRECTIVE
+        return self.access_paths, self.error_paths
 
-                    match = pattern_to_use.match(line)
-                    if not match:
-                        self.target.log.warning("Unexpected Apache log configuration: %s (%s)", line, path)
+    def _process_conf_file(self, path: Path) -> None:
```

Comment:
Maybe it's would be better to add either a `max_recurse` integer or to create a set of already processed file in order to prevent infinite recursion?